plugin/trace/trace.go

```diff
@@ -81,34 +89,29 @@ func (t *trace) Name() string { return "trace" }
 
 // ServeDNS implements the plugin.Handle interface.
 func (t *trace) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
-	trace := false
-	if t.every > 0 {
-		queryNr := atomic.AddUint64(&t.count, 1)
-
-		if queryNr%t.every == 0 {
-			trace = true
-		}
-	}
 	span := ot.SpanFromContext(ctx)
-	if !trace || span != nil {
+	if span != nil {
 		return plugin.NextOrFailure(t.Name(), t.Next, ctx, w, r)
 	}
 
 	req := request.Request{W: w, Req: r}
-	span = t.Tracer().StartSpan(spanName(ctx, req))
+	span = t.Tracer().StartSpan("servedns")
```

Comment:
Suggestion: `coredns.servedns` or `coredns.serve`.
We commonly prefix our operation name by the name of the software/framework/lib.

Reply:
Make sense.
Though the app itself already define which application those trace are coming from.
So i wonder if this is not redundant.

I will make the change and see what upstream says about it.